core/frontend/src/tile/map/WmtsCapabilities.ts

```diff
@@ -100,26 +100,29 @@ export namespace WmtsCapability {
     public readonly keywords?: string[];;
 
     constructor(json: any) {
-      this.abstract = json[OwsConstants.ABSTRACT_XMLTAG];
-      this.serviceType = json[OwsConstants.SERVICETYPE_XMLTAG];
-      this.serviceTypeVersion = json[OwsConstants.SERVICETYPEVERSION_XMLTAG];
-      this.title = json[OwsConstants.TITLE_XMLTAG];
+      this.abstract = json[OwsConstants.ABSTRACT_XMLTAG]?._text;
+      this.serviceType = json[OwsConstants.SERVICETYPE_XMLTAG]?._text;
+      this.serviceTypeVersion = json[OwsConstants.SERVICETYPEVERSION_XMLTAG]?._text;
+      this.title = json[OwsConstants.TITLE_XMLTAG]?._text;
 
       const keywords = json[OwsConstants.KEYWORDS_XMLTAG]?.[OwsConstants.KEYWORD_XMLTAG];
       if (keywords !== undefined) {
         this.keywords = [];
 
         if (Array.isArray(keywords)) {
           keywords.forEach((keyword: any) => {
-            this.keywords?.push(keyword);
+            if (keyword !== undefined) {
+              this.keywords?.push(keyword._text);
```

Comment:
The question mark after `this.keywords` is unnecessary, right? You set it to an empty array above.

Reply:
I agree with you, however typescript complaints if I remove it:
src/tile/map/WmtsCapabilities.ts:115:15 - error TS2532: Object is possibly 'undefined'.

Reply:
The compiler's assumptions about types don't extend into callbacks like the one you're passing to `forEach`. If you switch it to a `for` loop it should stop complaining.

Reply:
could replace it with a for loop over forEach and typescript should see since it's in the same scope that it can't be undefined. Or keep the forEach and use `!` to remove the runtime impact of the `?.`

Reply:
great thanks for the hints, just pushed a fix for it.